packages/kilt-did/src/Did.integration.spec.ts

```diff
@@ -0,0 +1,226 @@
+
+    await expect(queryById(id.address)).resolves.toBe(null)
+  }, 20_000)
+})
+
+it('creates and updates did', async () => {
+  const id = Identity.buildFromMnemonic('')
+
+  const tx = await createCreateTx(
+    {
+      didIdentifier: id.address,
+      keys: {
+        authentication: id.signKeyringPair,
+        encryption: { ...id.boxKeyPair, type: 'x25519' },
+      },
+      endpointUrl: 'https://example.com',
+    },
+    id.signKeyringPair
+  )
+
+  await expect(
+    BlockchainUtils.signAndSubmitTx(tx, alice, {
+      resolveOn: BlockchainUtils.IS_IN_BLOCK,
+    })
+  ).resolves.not.toThrow()
+
+  await expect(queryById(id.address)).resolves.toMatchObject<
+    Partial<IDidRecord>
+  >({
+    did: getDidFromIdentifier(id.address),
+    endpointUrl: 'https://example.com',
+  })
+
+  const tx2 = await createUpdateTx(
+    {
+      didIdentifier: id.address,
+      txCounter: 1,
+      newEndpointUrl: 'ftp://example.com/abc',
+    },
+    id.signKeyringPair
+  )
+
+  await expect(
+    BlockchainUtils.signAndSubmitTx(tx2, alice, {
+      resolveOn: BlockchainUtils.IS_IN_BLOCK,
+    })
+  ).resolves.not.toThrow()
+
+  await expect(queryById(id.address)).resolves.toMatchObject<
+    Partial<IDidRecord>
+  >({
+    did: getDidFromIdentifier(id.address),
+    endpointUrl: 'ftp://example.com/abc',
+  })
+}, 20_000)
+
+describe('did authorization', () => {
+  let id: IIdentity
+  beforeAll(async () => {
+    id = Identity.buildFromMnemonic('')
+    const tx = await createCreateTx(
+      {
+        didIdentifier: id.address,
+        keys: {
+          authentication: id.signKeyringPair,
+          encryption: { ...id.boxKeyPair, type: 'x25519' },
+          attestation: id.signKeyringPair,
+          delegation: id.signKeyringPair,
+        },
+        endpointUrl: 'https://example.com',
+      },
+      id.signKeyringPair
+    )
+    await expect(
+      BlockchainUtils.signAndSubmitTx(tx, alice, {
+        resolveOn: BlockchainUtils.IS_IN_BLOCK,
+      })
+    ).resolves.not.toThrow()
+
+    await expect(queryById(id.address)).resolves.toMatchObject<
+      Partial<IDidRecord>
+    >({
+      did: getDidFromIdentifier(id.address),
+    })
+  }, 20_000)
+
+  it('authorizes ctype creation with did signature', async () => {
+    const ctype = CType.fromSchema({
+      title: UUID.generate(),
+      properties: {},
+      type: 'object',
+      $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+    })
+    const call = await ctype.store()
+    const tx = await createAuthenticationTx(
+      {
+        didIdentifier: id.address,
+        txCounter: 1,
+        call,
+      },
+      id.signKeyringPair
+    )
+    await expect(
+      BlockchainUtils.signAndSubmitTx(tx, alice, {
+        resolveOn: BlockchainUtils.IS_IN_BLOCK,
+      })
+    ).resolves.not.toThrow()
+
+    await expect(ctype.verifyStored()).resolves.toEqual(true)
+  }, 20_000)
+
+  it('no longer authorizes ctype creation after did deactivation', async () => {
```

Comment:
I would call it DID deletion, not deactivation. And I would capitalize DID everywhere.